### ISSUE TITLE
Source map support

### ISSUE THREAD

**Issue by badeball** [badeball/cypress-cucumber-preprocessor#692]
Per today, source maps aren't / can't be utilized. EG. when multiple step definitions match, we throw an error

```
Error: Multiple matching step definitions for: a step
  a step
  /a step/
```

.. while cucumber-js on the other hand, would throw something like

```
Multiple step definitions match:
  a step - features/step_definitions/cli_steps.ts:44
  /a step/ - features/step_definitions/cli_steps.ts:49
```

Furthermore, we just output dummy object in the JSON reports.

```json
{
  "match": {
    "location": "not available:0"
  }
}
```

**Comment by badeball** [comment]
For reference, a PoC is avaliable in the branch `source-map`, but I'm waiting on https://github.com/evanw/esbuild/issues/2218 to see if there's a smoother way of handling paths.

**Comment by lmiller1990** [comment]
I'm guessing this is the same issue preventing this user from seeing the stack trace? https://github.com/cypress-io/cypress/issues/23563

**Comment by badeball** [comment]
No, this is an entirely separate issue.

**Comment by badeball** [comment]
This is solved with a6387f5, with documentation at https://github.com/badeball/cypress-cucumber-preprocessor/blob/master/docs/source-maps.md
